ext/ox/sax.c

```diff
@@ -984,7 +984,7 @@ static char read_element_end(SaxDrive dr) {
             } else {
                 snprintf(msg, sizeof(msg) - 1, "%selement '%s' closed but not opened", EL_MISMATCH, dr->buf.str);
                 ox_sax_drive_error_at(dr, msg, pos, line, col);
-                name = str2sym(dr, dr->buf.str, dr->buf.tail - dr->buf.str - 2, 0);
+                name = str2sym(dr, dr->buf.str, strlen(dr->buf.str), 0);
```

Comment:
As you pointed out, `dr->buf.str` may not be null terminated which will give an incorrect length. A different approach will be needed to handle when the end tag is empty if the existing code fails. It seems like it should work though.